### PR TITLE
feat(connections): update connections functionality

### PR DESCRIPTION
**Pull request by tplooker**
#### Short description of what this resolves:

Updates functionality associated to connection management.

#### Changes proposed in this pull request:

- Bumps AgentFramework version
- Adds publishing of a ConnectionsUpdated event when a new connection is made
- Adds support for deleting a connection

**Fixes**: #11 #12

### CODE REVIEW

File: src/Osma.Mobile.App/ViewModels/Connections/ConnectionsViewModel.cs [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 using System.Linq;
+using System.Reactive.Linq;
 using System.Threading.Tasks;
 using System.Windows.Input;
 using Acr.UserDialogs;
 using AgentFramework.Core.Contracts;
 using AgentFramework.Core.Messages.Connections;
 using AgentFramework.Core.Utils;
 using Autofac;
+using Osma.Mobile.App.Events.Osma.Mobile.App.Events;
```
Comment (Liam-Tait) [inline src/Osma.Mobile.App/ViewModels/Connections/ConnectionsViewModel.cs]:
```suggestion
using Osma.Mobile.App.Events;
```

Comment (Liam-Tait) [reply src/Osma.Mobile.App/ViewModels/Connections/ConnectionsViewModel.cs]: Can we remove this Nested namespace?

File: src/Osma.Mobile.App/Events/ApplicationEvent.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Osma.Mobile.App.Events
+{
+    namespace Osma.Mobile.App.Events
```
Comment (Liam-Tait) [inline src/Osma.Mobile.App/Events/ApplicationEvent.cs]: Looks like we have some nested namespaces here, can we remove them?

File: src/Osma.Mobile.App/ViewModels/Connections/AcceptInviteViewModel.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@
 using Acr.UserDialogs;
 using AgentFramework.Core.Contracts;
 using AgentFramework.Core.Messages.Connections;
+using Osma.Mobile.App.Events.Osma.Mobile.App.Events;
```
Comment (Liam-Tait) [inline src/Osma.Mobile.App/ViewModels/Connections/AcceptInviteViewModel.cs]: Nested namespaces
```suggestion
using Osma.Mobile.App.Events;
```

File: src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs [hunk 4]
```diff
@@ -3,7 +3,9 @@
 using System.Threading.Tasks;
 using System.Windows.Input;
 using Acr.UserDialogs;
+using AgentFramework.Core.Contracts;
 using AgentFramework.Core.Models.Records;
+using Osma.Mobile.App.Events.Osma.Mobile.App.Events;
```
Comment (Liam-Tait) [inline src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs]:
```suggestion
using Osma.Mobile.App.Events;
```

File: src/Osma.Mobile.App/Events/ApplicationEvent.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        public enum ApplicationEventType
+        {
+            ConnectionsUpdated
+        }
+
+        public class ApplicationEvent
+        {
+            public ApplicationEventType EventType { get; set; }
```
Comment (Liam-Tait) [inline src/Osma.Mobile.App/Events/ApplicationEvent.cs]: Is it useful to duplicate the name `Event` here? The item is an event, could we rename this from `EventType` to `Type`
```suggestion
            public ApplicationEventType Type { get; set; }
```

File: src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs [hunk 6]
```diff
@@ -74,13 +87,24 @@ public Task RefreshTransactions()
...
 
         public ICommand DeleteConnectionCommand => new Command(async () =>
         {
-            // TODO OS-218 Delete the connection
+            var dialog = DialogService.Loading("Deleting");
+
+            var context = await _agentContextProvider.GetContextAsync();
+            await _connectionService.DeleteAsync(context, _record.Id);
+
+            _eventAggregator.Publish(new ApplicationEvent() { EventType = ApplicationEventType.ConnectionsUpdated });
```
Comment (Liam-Tait) [inline src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs]: Thoughts on publishing the event from inside connectionService.DeleteAsync instead?

File: src/Osma.Mobile.App/ViewModels/Connections/ConnectionsViewModel.cs [hunk 7]
```diff
@@ -21,23 +23,31 @@ public class ConnectionsViewModel : ABaseViewModel
...
             _agentContextProvider = agentContextProvider;
+            _eventAggregator = eventAggregator;
             _scope = scope;
         }
 
         public override async Task InitializeAsync(object navigationData)
         {
             await RefreshConnections();
+
+            _eventAggregator.GetEventByType<ApplicationEvent>()
```
Comment (Liam-Tait) [inline src/Osma.Mobile.App/ViewModels/Connections/ConnectionsViewModel.cs:47]: Nice, I like the pattern

**Review by Liam-Tait — CHANGES_REQUESTED** [review]

File: src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs (hunk 6 above)
Comment (tplooker) [reply src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs]: The argument could be made that we need more events at crucial points emitted by the framework, @tmarkovski your thoughts about a more prolific usage of events inside the framework for cases such as this? In any case these would not be application events instead framework or service events. My personal thoughts are it doesnt feel too onerous for the application implementer having to write their own events over the framework and that it currently allows them to maintain good control over the event logic and structure.

**Review by Liam-Tait — APPROVED** [review]

**Pull request merged**